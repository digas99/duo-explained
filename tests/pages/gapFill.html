<!DOCTYPE html>
<html data-duo-theme="dark" dir="ltr" lang="en"><head><title data-react-helmet="true">Duolingo - The world's best way to learn French</title>     </head><body style="--onetrust-banner-height: 0px;"><div class="fs-unmask" id="root" onclick=""><div class="kPqwA _2kkzG"><div class="_3yE3H"><div class="_2G9ED _29gfw"><div class="wqSzE"><div class="_3v4ux"><div class="I-Avc _1zcW8"><button class="_3vGNs _2YF0P _3X5b9" data-test="quit-button"><img class="_3X5b9 _1o-YO" src="./gapFill_files/df223d5b9feb8017b323ed21103eb5ac.svg"/></button><div aria-valuemax="1" aria-valuemin="0" aria-valuenow="0.11764705882352941" class="_358Nq" role="progressbar" style="--web-ui_progress-bar-color: rgb(var(--color-owl)); --web-ui_progress-bar-shine-height: 3px; --__internal__progress-bar-height: 16px; --__internal__progress-bar-inner-value: 0%; --__internal__progress-bar-value: 11.76%;"><div class="_2-LJi"><div class="_2fuvy" style="opacity: 1;"><div class="_3Dzp2 _2fuvy"></div><div class="_1EzVJ"></div></div><div class="_3FEy7" style="opacity: 1;"><span class="_27WD7 _1ZZav">2 in a row</span><div class="_2fYtx"><span style="--progress-bar-height: 16px;"><span class="RxQcJ fs-exclude SVDhH"></span></span></div></div></div></div><div class="_3ww7z"><img src="./gapFill_files/0c1a523b4b5882a97e4df162f4b5c58b.svg" style="height: 32px; width: 32px;"/><img src="./gapFill_files/8d4c5fcd9cedabfd155d3eda8af269bc.svg"/></div></div></div></div><div class="RMEuZ _1GVfY"><div class="_3GuWo _1cTBC"><div class="_1XNQX _3V6my"><div class="_1fxa4 _1Mopf" data-test="challenge challenge-gapFill"><div class="_2n5fx _1JTA4 _1es1Y K3vbJ"><div class="U41Ye"><h1 class="_3EOK0" data-test="challenge-header"><span>Fill in the blank</span></h1></div><div class="_3Jh10 _1IiFg f7WE2 _3rat3"><div class="_3gSoe _35mGI"><div dir="ltr"><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">B</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">o</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">n</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">j</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">o</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">u</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">r</span><div aria-label="Bonjour" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 64.1562px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">M</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">a</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">r</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">c</span><div aria-label="Marc" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 40.625px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">,</span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">c</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">'</span><div aria-label="c'" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 12.7344px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">e</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">s</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">t</span><div aria-label="est" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 25.0938px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">m</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">o</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">i</span><div aria-label="moi" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 30.5938px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">,</span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">E</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">l</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">s</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">a</span><div aria-label="Elsa" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 35.4062px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">;</span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_3ykJt XxgPa" style="background: none;">j</span><span aria-hidden="true" class="_3ykJt XxgPa" style="background: none;">e</span><div aria-label="je" class="tFegI _3ykJt XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 14.4531px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_3ykJt XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_3ykJt XxgPa" style="background: none;">s</span><span aria-hidden="true" class="_3ykJt XxgPa" style="background: none;">u</span><span aria-hidden="true" class="_3ykJt XxgPa" style="background: none;">i</span><span aria-hidden="true" class="_3ykJt XxgPa" style="background: none;">s</span><div aria-label="suis" class="tFegI _3ykJt XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 33.5938px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_3ykJt XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_3ykJt XxgPa" style="background: none;">i</span><span aria-hidden="true" class="_3ykJt XxgPa" style="background: none;">c</span><span aria-hidden="true" class="_3ykJt XxgPa" style="background: none;">i</span><div aria-label="ici" class="tFegI _3ykJt XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 18.4062px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">,</span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">d</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">e</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">v</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">a</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">n</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">t</span><div aria-label="devant" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 55.1719px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">t</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">a</span><div aria-label="ta" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 15.9375px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;"> </span></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">m</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">a</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">i</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">s</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">o</span><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">n</span><div aria-label="maison" class="tFegI _2IGwo XxgPa _33wyM" data-test="hint-token" style="height: 27px; left: 0px; top: 0px; width: 60.1406px;" tabindex="0"></div></span><span class="_5HFLU" lang="fr"><span aria-hidden="true" class="_2IGwo XxgPa" style="background: none;">.</span></span></div></div><div aria-label="choice" class="PaKCO _1vQbM _1BTbT ccJ7o" role="radiogroup"><div aria-checked="false" aria-disabled="true" class="-XCTI _2R-aH _1VWfn _3YXhY ufykF" data-test="challenge-choice" role="radio" tabindex="-1"><span class="_3zbIX _20AkF _3U02j">1</span><span class="lEvgJ _3r1Ls" data-test="challenge-judge-text" dir="ltr" lang="fr">je suis professeur<span class="_1QC2f" style="transform: scale(-1, 1);"><span class="RxQcJ fs-exclude FMPra"></span><span class="RxQcJ fs-exclude _2iWiH"></span></span></span></div><div aria-checked="false" aria-disabled="true" class="-XCTI _2R-aH _1VWfn _3YXhY ufykF" data-test="challenge-choice" role="radio" tabindex="-1"><span class="_3zbIX _20AkF _3U02j">2</span><span class="lEvgJ _3r1Ls" data-test="challenge-judge-text" dir="ltr" lang="fr">j'ai un lit<span class="_1QC2f" style="transform: scale(-1, -1);"><span class="RxQcJ fs-exclude FMPra"></span><span class="RxQcJ fs-exclude _2iWiH"></span></span></span></div><div aria-checked="true" aria-disabled="true" class="-XCTI _3gk8O _2R-aH _1VWfn HsrFv ufykF _3YXhY ufykF" data-test="challenge-choice" role="radio" tabindex="-1"><span class="_3TbFR _20AkF _3U02j">3</span><span class="lEvgJ _3r1Ls" data-test="challenge-judge-text" dir="ltr" lang="fr">je suis ici<span class="_1QC2f" style="transform: scale(-1, 1);"><span class="RxQcJ fs-exclude FMPra"></span><span class="RxQcJ fs-exclude _2iWiH"></span></span></span></div></div></div></div></div></div></div></div><div class="_3GuWo _1cTBC _1QQhE"><div class="_1XNQX"><div class="_3FB5S _1VTif _3yMvO _2HXQ9 d-cgpt-footer" id="session/PlayerFooter"><div class="U8jH3 jHbiF"><div class="AXBVw _2Ycbe _8wIx-"><div class="_1k6eg" data-test="blame blame-correct"><div class="_31ixh HJdJI fwpkb"><img alt="" class="_3oCTd" src="./gapFill_files/b377ec812acb8c96d87d52e8009478ad.svg"/></div><div class="d6Xhl"><div class="_3LKiy"><div class="_1D3fo"><h2 class="_2U7Gm WXwlk">Correct!</h2><div class="_2jz5U WXwlk" dir="ltr" lang="en">Meaning: Hello Marc, it's me, Elsa; I'm here, in front of your house.</div></div></div><div class="_2yBiG"><div class="_10y_e"><button class="_1pDd6 _2JTJI bafGS _2LoNU VzbUl _1saKQ _1AgKJ"><span class="_24Rh2"><span class="_2e3N3 _3jt-c"></span><span class="_1qfbO _2Rt1l">Report</span></span></button></div></div></div></div></div><div class="MYehf d-cgpt-explain-button-wrapper"><button aria-disabled="false" class="_1x5JY _1M9iF _36g4N _2YF0P _3DbUj _38g3s _2oGJR d-cgpt-explain-button d-cgpt-explain-button-enabled" data-hasclick="true" id="d-cgpt-explain-button"><span class="_1o-YO">Explain</span></button><button aria-disabled="false" class="_1x5JY _1M9iF _36g4N _2YF0P _3DbUj _38g3s _2oGJR" data-test="player-next"><span class="_1o-YO">Continue</span></button></div></div></div>
<div class="d-cgpt-speech-bubble" style="bottom: 141px; display: none;">
<div>
<textarea placeholder="If you want, add to the query to ChatGPT..."></textarea>
<div class="d-cgpt-speech-bubble-tip"></div>
</div>
</div>
</div></div></div></div></div><div class="Vm8CO _1Fnem" data-test="drawer-backdrop" style="pointer-events: none;"></div></div><div class="fs-unmask" id="overlays" onclick=""><div class="Vm8CO _2zxQ8" style="pointer-events: none;"></div><a class="_1Qh5D _36g4N _2YF0P _3dDzz _1wiIJ" href="https://preview.duolingo.com/help/support-request" rel="noopener" target="_blank" title="Report feedback"><img class="_3QUr-" src="./gapFill_files/b23d90ca7ea01eb330158d4939dc4899.svg"/></a><div data-focus-guard="true" style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;" tabindex="-1"></div><div data-focus-lock-disabled="disabled"></div><div data-focus-guard="true" style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;" tabindex="-1"></div></div><div><div class="grecaptcha-badge" data-style="bottomright" style="width: 256px; height: 60px; display: block; transition: right 0.3s; position: fixed; bottom: 14px; right: -186px; box-shadow: gray 0px 0px 5px; border-radius: 2px; overflow: hidden;"><div class="grecaptcha-logo"></div><div class="grecaptcha-error"></div><textarea class="g-recaptcha-response" id="g-recaptcha-response-100000" name="g-recaptcha-response" style="width: 250px; height: 40px; border: 1px solid rgb(193, 193, 193); margin: 10px 25px; padding: 0px; resize: none; display: none;"></textarea></div></div><div id="onetrust-consent-sdk"><div class="onetrust-pc-dark-filter ot-hide ot-fade-in"></div></div></body></html>