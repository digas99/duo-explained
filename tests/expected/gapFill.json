{
	"content": {
		"sentence": "Bonjour Marc, c'est moi, Elsa ; <blank>, devant ta maison.",
		"userAnswer": "Bonjour Marc, c'est moi, Elsa ; je suis ici, devant ta maison.",
		"options": [
			{
				"option": 1,
				"choices": [
					"je suis professeur"
				]
			},
			{
				"option": 2,
				"choices": [
					"j'ai un lit"
				]
			},
			{
				"option": 3,
				"choices": [
					"je suis ici"
				]
			}
		]
	},
	"question": "Fill in the blank",
	"wrapper": {},
	"type": "gapFill"
}